### PR TITLE
✨ NEW: Restart and Run All buttons

### PR DESCRIPTION
**Pull request by rowanc1**
# Overview
This adds functionality to restart and run the kernel. It clears the outputs of each cell with a `"Waiting for kernel ... "` *before* the kernel is restarted (a slight difference from jupyter) but this is helpful as there is no other UI indicator that the button worked.

thebe-restart-and-runall

## Updates

* I have changed the return of `renderCell` to return `{cell, execute, setOutputText}` which allows you to use them in other contexts.
* There is now a `thebelab.cells` which is an array of the above.

See #237

### CODE REVIEW

**Comment by choldgraf** [comment]
This looks good to me! Note that I added another cell to the `development.html` example in order to be able to demo this locally (which worked!), and then when I pushed to your branch github informed me that apparently another example has already been added to that file on `master`, so I just merged in that part of the master branch 👍

**Comment by choldgraf** [comment]
we probably want some kind of indication to users that a kernel has been restarted. Maybe we clear the outputs or put a little "loading" SVG or something like that. (doesn't need to be in this PR)

**Comment by rowanc1** [comment]
Ha yeah, I was seeing @minrk working on that.

Do you think this is good to merge in?

File: src/thebelab.js [hunk 1]
```diff
@@ -237,10 +247,24 @@ function renderCell(element, options) {
   function restart() {
     let kernel = $cell.data("kernel");
     if (kernel) {
-      kernelPromise.then((kernel) => {
-        kernel.restart();
+      return kernelPromise.then(async (kernel) => {
+        await kernel.restart();
```
Comment (choldgraf) [inline src/thebelab.js:251]:
```suggestion
        setOutputText("Restarting kernel...")
        await kernel.restart();
```

is adding an indication for a restarting kernel as simple as this?

Comment (rowanc1) [reply src/thebelab.js:251]: That will only work for the current cell I believe.

**Comment by choldgraf** [comment]
this looks good to me and testing it locally worked great, so I'll merge this in and we can continue iterating on the buttons / layout / etc in subsequent issues! Thanks @rowanc1 ✨

**Pull request merged**